src/fast_layers.jl

```diff
@@ -96,9 +104,8 @@ ZygoteRules.@adjoint function (f::FastDense)(x,p)
   end
   y,FastDense_adjoint
 end
-paramlength(f::FastDense) = f.out*(f.in + 1)
+paramlength(f::FastDense) = length(f.initial_params())

```

Comment:
revert and fix this: generating a whole parameter vector to check the length is very inefficient.